### ISSUE TITLE
Fix flaky integ tests

### ISSUE THREAD

**Issue by ohltyler** [opensearch-project/index-management#258]
When running ISM integ tests in Jenkins, seeing various failures related to missing test snapshots, system indices, etc. It is likely that the cluster itself is under-resourced and is contributing to these failures as well, but for test hardening, they should be improved with having longer timeouts before failing out.

List of failed tests + traces (version 1.2.2, security enabled, Jenkins job `ohltyler-integ-test-stable #69`):
```
2022-02-01 16:32:22 INFO     ===============================================
2022-02-01 16:32:22 INFO     Running integration tests for index-management
2022-02-01 16:32:22 INFO     ===============================================
2022-02-01 16:32:22 INFO     Executing "/var/jenkins/workspace/Playground/ohltyler-integ-test-stable/scripts/default/integtest.sh -b localhost -p 9200 -s true -v 1.2.2" in /tmp/tmp0q50wr1x/index-management
2022-02-01 16:51:28 INFO     Recording component test results for index-management at /var/jenkins/workspace/Playground/ohltyler-integ-test-stable/test-results/901d4ae0e9e6486db8917e3dc4b4f07b/integ-test/index-management/with-security/test-results
2022-02-01 16:51:28 INFO     Integration test run failed for component index-management
2022-02-01 16:51:28 INFO     WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.jetbrains.kotlin.com.intellij.util.ReflectionUtil (file:/home/ec2-user/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-compiler-embeddable/1.4.31/6451ea797cef544e81f507b0d9959cd97ae09c0/kotlin-compiler-embeddable-1.4.31.jar) to method java.util.ResourceBundle.setParent(java.util.ResourceBundle)
WARNING: Please consider reporting this to the maintainers of org.jetbrains.kotlin.com.intellij.util.ReflectionUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
 
REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test snapshot repository blocked" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=es-CO -Dtests.timezone=Atlantic/Faroe -Druntime.java=14
 
REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test successful wait for snapshot step - empty snapshot name" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=es-CO -Dtests.timezone=Atlantic/Faroe -Druntime.java=14
 
REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test basic with templated snapshot name" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=es-CO -Dtests.timezone=Atlantic/Faroe -Druntime.java=14
 
REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test failed wait for snapshot step" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=es-CO -Dtests.timezone=Atlantic/Faroe -Druntime.java=14
 
REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test successful wait for snapshot step" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=es-CO -Dtests.timezone=Atlantic/Faroe -Druntime.java=14
 
REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test basic" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=es-CO -Dtests.timezone=Atlantic/Faroe -Druntime.java=14
 
REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test basic with invalid templated snapshot name default to indexName" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=es-CO -Dtests.timezone=Atlantic/Faroe -Druntime.java=14
 
 
Suite: Test class org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT
  2> REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test snapshot repository blocked" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=es-CO -Dtests.timezone=Atlantic/Faroe -Druntime.java=14
  2> org.opensearch.client.ResponseException: method [PUT], host [https://localhost:9200], URI [_snapshot/hello-world], status line [HTTP/1.1 500 Internal Server Error]
    {"error":{"root_cause":[{"type":"repository_verification_exception","reason":"[hello-world] [[teCvyQDsTdiPHZVi1OKq8A, 'RemoteTransportException[[ip-10-0-25-235.ec2.internal][127.0.0.1:9301][internal:admin/repository/verify]]; nested: RepositoryMissingException[[hello-world] missing];']]"}],"type":"repository_verification_exception","reason":"[hello-world] [[teCvyQDsTdiPHZVi1OKq8A, 'RemoteTransportException[[ip-10-0-25-235.ec2.internal][127.0.0.1:9301][internal:admin/repository/verify]]; nested: RepositoryMissingException[[hello-world] missing];']]"},"status":500}
        at __randomizedtesting.SeedInfo.seed([FC0793C0602FDFDE:97699C24773E4A1C]:0)
        at org.opensearch.client.RestClient.convertResponse(RestClient.java:344)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:314)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:289)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest(TestHelpers.kt:106)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest$default(TestHelpers.kt:95)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase.createRepository(IndexStateManagementRestTestCase.kt:628)
        at org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test snapshot repository blocked(SnapshotActionIT.kt:319)
  2> REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test successful wait for snapshot step - empty snapshot name" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=es-CO -Dtests.timezone=Atlantic/Faroe -Druntime.java=14
  2> org.opensearch.client.ResponseException: method [PUT], host [https://localhost:9200], URI [_snapshot/repository], status line [HTTP/1.1 500 Internal Server Error]
    {"error":{"root_cause":[{"type":"repository_verification_exception","reason":"[repository] [[teCvyQDsTdiPHZVi1OKq8A, 'RemoteTransportException[[ip-10-0-25-235.ec2.internal][127.0.0.1:9301][internal:admin/repository/verify]]; nested: RepositoryMissingException[[repository] missing];']]"}],"type":"repository_verification_exception","reason":"[repository] [[teCvyQDsTdiPHZVi1OKq8A, 'RemoteTransportException[[ip-10-0-25-235.ec2.internal][127.0.0.1:9301][internal:admin/repository/verify]]; nested: RepositoryMissingException[[repository] missing];']]"},"status":500}
        at __randomizedtesting.SeedInfo.seed([FC0793C0602FDFDE:B9C7C5FB2FF9BB21]:0)
        at org.opensearch.client.RestClient.convertResponse(RestClient.java:344)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:314)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:289)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest(TestHelpers.kt:106)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest$default(TestHelpers.kt:95)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase.createRepository(IndexStateManagementRestTestCase.kt:628)
        at org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test successful wait for snapshot step - empty snapshot name(SnapshotActionIT.kt:217)
  2> REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test basic with templated snapshot name" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=es-CO -Dtests.timezone=Atlantic/Faroe -Druntime.java=14
  2> org.opensearch.client.ResponseException: method [PUT], host [https://localhost:9200], URI [_snapshot/repository], status line [HTTP/1.1 500 Internal Server Error]
    {"error":{"root_cause":[{"type":"repository_verification_exception","reason":"[repository] [[teCvyQDsTdiPHZVi1OKq8A, 'RemoteTransportException[[ip-10-0-25-235.ec2.internal][127.0.0.1:9301][internal:admin/repository/verify]]; nested: RepositoryMissingException[[repository] missing];']]"}],"type":"repository_verification_exception","reason":"[repository] [[teCvyQDsTdiPHZVi1OKq8A, 'RemoteTransportException[[ip-10-0-25-235.ec2.internal][127.0.0.1:9301][internal:admin/repository/verify]]; nested: RepositoryMissingException[[repository] missing];']]"},"status":500}
        at __randomizedtesting.SeedInfo.seed([FC0793C0602FDFDE:36E9F595D114CC20]:0)
        at org.opensearch.client.RestClient.convertResponse(RestClient.java:344)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:314)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:289)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest(TestHelpers.kt:106)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest$default(TestHelpers.kt:95)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase.createRepository(IndexStateManagementRestTestCase.kt:628)
        at org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test basic with templated snapshot name(SnapshotActionIT.kt:93)
  2> REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test failed wait for snapshot step" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=es-CO -Dtests.timezone=Atlantic/Faroe -Druntime.java=14
  2> org.opensearch.client.ResponseException: method [PUT], host [https://localhost:9200], URI [_snapshot/repository], status line [HTTP/1.1 500 Internal Server Error]
    {"error":{"root_cause":[{"type":"repository_verification_exception","reason":"[repository] [[teCvyQDsTdiPHZVi1OKq8A, 'RemoteTransportException[[ip-10-0-25-235.ec2.internal][127.0.0.1:9301][internal:admin/repository/verify]]; nested: RepositoryMissingException[[repository] missing];']]"}],"type":"repository_verification_exception","reason":"[repository] [[teCvyQDsTdiPHZVi1OKq8A, 'RemoteTransportException[[ip-10-0-25-235.ec2.internal][127.0.0.1:9301][internal:admin/repository/verify]]; nested: RepositoryMissingException[[repository] missing];']]"},"status":500}
        at __randomizedtesting.SeedInfo.seed([FC0793C0602FDFDE:8C124B94C036CC26]:0)
        at org.opensearch.client.RestClient.convertResponse(RestClient.java:344)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:314)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:289)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest(TestHelpers.kt:106)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest$default(TestHelpers.kt:95)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase.createRepository(IndexStateManagementRestTestCase.kt:628)
        at org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test failed wait for snapshot step(SnapshotActionIT.kt:266)
  2> REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test successful wait for snapshot step" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=es-CO -Dtests.timezone=Atlantic/Faroe -Druntime.java=14
  2> org.opensearch.client.ResponseException: method [PUT], host [https://localhost:9200], URI [_snapshot/repository], status line [HTTP/1.1 500 Internal Server Error]
    {"error":{"root_cause":[{"type":"repository_verification_exception","reason":"[repository] [[teCvyQDsTdiPHZVi1OKq8A, 'RemoteTransportException[[ip-10-0-25-235.ec2.internal][127.0.0.1:9301][internal:admin/repository/verify]]; nested: RepositoryMissingException[[repository] missing];']]"}],"type":"repository_verification_exception","reason":"[repository] [[teCvyQDsTdiPHZVi1OKq8A, 'RemoteTransportException[[ip-10-0-25-235.ec2.internal][127.0.0.1:9301][internal:admin/repository/verify]]; nested: RepositoryMissingException[[repository] missing];']]"},"status":500}
        at __randomizedtesting.SeedInfo.seed([FC0793C0602FDFDE:49EA18D2B1D3F30]:0)
        at org.opensearch.client.RestClient.convertResponse(RestClient.java:344)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:314)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:289)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest(TestHelpers.kt:106)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest$default(TestHelpers.kt:95)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase.createRepository(IndexStateManagementRestTestCase.kt:628)
        at org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test successful wait for snapshot step(SnapshotActionIT.kt:168)
  2> REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test basic" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=es-CO -Dtests.timezone=Atlantic/Faroe -Druntime.java=14
  2> org.opensearch.client.ResponseException: method [PUT], host [https://localhost:9200], URI [_snapshot/repository], status line [HTTP/1.1 500 Internal Server Error]
    {"error":{"root_cause":[{"type":"repository_verification_exception","reason":"[repository] [[teCvyQDsTdiPHZVi1OKq8A, 'RemoteTransportException[[ip-10-0-25-235.ec2.internal][127.0.0.1:9301][internal:admin/repository/verify]]; nested: RepositoryMissingException[[repository] missing];']]"}],"type":"repository_verification_exception","reason":"[repository] [[teCvyQDsTdiPHZVi1OKq8A, 'RemoteTransportException[[ip-10-0-25-235.ec2.internal][127.0.0.1:9301][internal:admin/repository/verify]]; nested: RepositoryMissingException[[repository] missing];']]"},"status":500}
        at __randomizedtesting.SeedInfo.seed([FC0793C0602FDFDE:EF2C1F2A46BE3B0A]:0)
        at org.opensearch.client.RestClient.convertResponse(RestClient.java:344)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:314)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:289)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest(TestHelpers.kt:106)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest$default(TestHelpers.kt:95)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase.createRepository(IndexStateManagementRestTestCase.kt:628)
        at org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test basic(SnapshotActionIT.kt:56)
  2> REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test basic with invalid templated snapshot name default to indexName" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=es-CO -Dtests.timezone=Atlantic/Faroe -Druntime.java=14
  2> org.opensearch.client.ResponseException: method [PUT], host [https://localhost:9200], URI [_snapshot/repository], status line [HTTP/1.1 500 Internal Server Error]
    {"error":{"root_cause":[{"type":"repository_verification_exception","reason":"[repository] [[teCvyQDsTdiPHZVi1OKq8A, 'RemoteTransportException[[ip-10-0-25-235.ec2.internal][127.0.0.1:9301][internal:admin/repository/verify]]; nested: RepositoryMissingException[[repository] missing];']]"}],"type":"repository_verification_exception","reason":"[repository] [[teCvyQDsTdiPHZVi1OKq8A, 'RemoteTransportException[[ip-10-0-25-235.ec2.internal][127.0.0.1:9301][internal:admin/repository/verify]]; nested: RepositoryMissingException[[repository] missing];']]"},"status":500}
        at __randomizedtesting.SeedInfo.seed([FC0793C0602FDFDE:FAEA4C2105828994]:0)
        at org.opensearch.client.RestClient.convertResponse(RestClient.java:344)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:314)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:289)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest(TestHelpers.kt:106)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest$default(TestHelpers.kt:95)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase.createRepository(IndexStateManagementRestTestCase.kt:628)
        at org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT.test basic with invalid templated snapshot name default to indexName(SnapshotActionIT.kt:130)
  2> NOTE: leaving temporary files on disk at: /tmp/tmp0q50wr1x/index-management/build/testrun/integTest/temp/org.opensearch.indexmanagement.indexstatemanagement.action.SnapshotActionIT_FC0793C0602FDFDE-001
  2> NOTE: test params are: codec=Asserting(Lucene87): {}, docValues:{}, maxPointsInLeafNode=1144, maxMBSortInHeap=6.7540667821707725, sim=Asserting(RandomSimilarity(queryNorm=false): {}), locale=es-CO, timezone=Atlantic/Faroe
  2> NOTE: Linux 4.14.243-185.433.amzn2.x86_64 amd64/AdoptOpenJDK 14.0.2 (64-bit)/cpus=16,threads=1,free=258807480,total=536870912
  2> NOTE: All tests run in this JVM: [ManagedIndexRunnerIT, ActionRetryIT, ActionTimeoutIT, AllocationActionIT, CloseActionIT, ForceMergeActionIT, IndexPriorityActionIT, IndexStateManagementHistoryIT, OpenActionIT, ReadOnlyActionIT, ReadWriteActionIT, ReplicaCountActionIT, RolloverActionIT, RollupActionIT, SnapshotActionIT]
 
REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.resthandler.IndexStateManagementRestApiIT.test deleting a policy" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=en-PH -Dtests.timezone=MET -Druntime.java=14
 
 
Suite: Test class org.opensearch.indexmanagement.indexstatemanagement.resthandler.IndexStateManagementRestApiIT
  2> REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.resthandler.IndexStateManagementRestApiIT.test deleting a policy" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=en-PH -Dtests.timezone=MET -Druntime.java=14
  2> java.lang.AssertionError: 2 active tasks found:
    internal:index/shard/recovery/start_recovery 0mS-1HDBR9WqCHI3i1VJPw:52459 -                            transport 1643733921147 16:45:21 57.8ms      127.0.0.1 ip-10-0-25-235.ec2.internal 
    internal:index/shard/recovery/finalize       teCvyQDsTdiPHZVi1OKq8A:4909  -                            transport 1643733921203 16:45:21 2ms         127.0.0.1 ip-10-0-25-235.ec2.internal 
     expected:<0> but was:<2>
        at __randomizedtesting.SeedInfo.seed([FC0793C0602FDFDE:1F2AFF6DAADDB186]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:647)
        at org.opensearch.test.rest.OpenSearchRestTestCase.lambda$waitForPendingTasks$3(OpenSearchRestTestCase.java:420)
        at org.opensearch.test.OpenSearchTestCase.assertBusy(OpenSearchTestCase.java:1048)
        at org.opensearch.test.rest.OpenSearchRestTestCase.waitForPendingTasks(OpenSearchRestTestCase.java:393)
        at org.opensearch.test.rest.OpenSearchRestTestCase.waitForPendingTasks(OpenSearchRestTestCase.java:380)
        at org.opensearch.indexmanagement.ODFERestTestCase$waitForCleanup$1.invoke(ODFERestTestCase.kt:80)
        at org.opensearch.indexmanagement.ODFERestTestCase$waitForCleanup$1.invoke(ODFERestTestCase.kt:51)
        at org.opensearch.indexmanagement.TestHelpersKt.waitFor(TestHelpers.kt:140)
        at org.opensearch.indexmanagement.TestHelpersKt.waitFor$default(TestHelpers.kt:134)
        at org.opensearch.indexmanagement.ODFERestTestCase.waitForCleanup(ODFERestTestCase.kt:77)
        at jdk.internal.reflect.GeneratedMethodAccessor14.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
        at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
        at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
        at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
        at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
        at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
        at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
        at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
        at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
        at java.base/java.lang.Thread.run(Thread.java:832)
  2> NOTE: leaving temporary files on disk at: /tmp/tmp0q50wr1x/index-management/build/testrun/integTest/temp/org.opensearch.indexmanagement.indexstatemanagement.resthandler.IndexStateManagementRestApiIT_FC0793C0602FDFDE-001
  2> NOTE: test params are: codec=Asserting(Lucene87), sim=Asserting(RandomSimilarity(queryNorm=true): {}), locale=en-PH, timezone=MET
  2> NOTE: Linux 4.14.243-185.433.amzn2.x86_64 amd64/AdoptOpenJDK 14.0.2 (64-bit)/cpus=16,threads=1,free=251798960,total=536870912
  2> NOTE: All tests run in this JVM: [ManagedIndexRunnerIT, ActionRetryIT, ActionTimeoutIT, AllocationActionIT, CloseActionIT, ForceMergeActionIT, IndexPriorityActionIT, IndexStateManagementHistoryIT, OpenActionIT, ReadOnlyActionIT, ReadWriteActionIT, ReplicaCountActionIT, RolloverActionIT, RollupActionIT, SnapshotActionIT, TransitionActionIT, ISMTemplateRestAPIIT, IndexStateManagementRestApiIT]
 
REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestAddPolicyActionIT.test index list" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=da -Dtests.timezone=Asia/Choibalsan -Druntime.java=14
 
 
Suite: Test class org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestAddPolicyActionIT
  2> REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestAddPolicyActionIT.test index list" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=da -Dtests.timezone=Asia/Choibalsan -Druntime.java=14
  2> org.opensearch.client.ResponseException: method [POST], host [https://localhost:9200], URI [/_opendistro/_ism/add/movies_2], status line [HTTP/1.1 404 Not Found]
    {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [.opendistro-ism-config]","index":".opendistro-ism-config","resource.id":".opendistro-ism-config","resource.type":"index_expression","index_uuid":"_na_"}],"type":"index_not_found_exception","reason":"no such index [.opendistro-ism-config]","index":".opendistro-ism-config","resource.id":".opendistro-ism-config","resource.type":"index_expression","index_uuid":"_na_"},"status":404}
        at __randomizedtesting.SeedInfo.seed([FC0793C0602FDFDE:1B33A1AC3F28BBAC]:0)
        at org.opensearch.client.RestClient.convertResponse(RestClient.java:344)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:314)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:289)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest(TestHelpers.kt:130)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest$default(TestHelpers.kt:120)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase.addPolicyToIndex(IndexStateManagementRestTestCase.kt:249)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase.createIndex(IndexStateManagementRestTestCase.kt:206)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase.createIndex$default(IndexStateManagementRestTestCase.kt:190)
        at org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestAddPolicyActionIT.test index list(RestAddPolicyActionIT.kt:125)
  2> NOTE: leaving temporary files on disk at: /tmp/tmp0q50wr1x/index-management/build/testrun/integTest/temp/org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestAddPolicyActionIT_FC0793C0602FDFDE-001
  2> NOTE: test params are: codec=Asserting(Lucene87): {}, docValues:{}, maxPointsInLeafNode=110, maxMBSortInHeap=7.495748395182151, sim=Asserting(RandomSimilarity(queryNorm=false): {}), locale=da, timezone=Asia/Choibalsan
  2> NOTE: Linux 4.14.243-185.433.amzn2.x86_64 amd64/AdoptOpenJDK 14.0.2 (64-bit)/cpus=16,threads=1,free=353092688,total=536870912
  2> NOTE: All tests run in this JVM: [ManagedIndexRunnerIT, ActionRetryIT, ActionTimeoutIT, AllocationActionIT, CloseActionIT, ForceMergeActionIT, IndexPriorityActionIT, IndexStateManagementHistoryIT, OpenActionIT, ReadOnlyActionIT, ReadWriteActionIT, ReplicaCountActionIT, RolloverActionIT, RollupActionIT, SnapshotActionIT, TransitionActionIT, ISMTemplateRestAPIIT, IndexStateManagementRestApiIT, RestAddPolicyActionIT]
 
REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestChangePolicyActionIT.test change policy API should only apply to indices in the state filter" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=sr-Latn-BA -Dtests.timezone=Etc/GMT+12 -Druntime.java=14
 
REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestChangePolicyActionIT.test starting from a specific state using change policy API" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=sr-Latn-BA -Dtests.timezone=Etc/GMT+12 -Druntime.java=14
 
REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestChangePolicyActionIT.test allowing change policy to happen in middle of state if same state structure" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=sr-Latn-BA -Dtests.timezone=Etc/GMT+12 -Druntime.java=14
 
 
Suite: Test class org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestChangePolicyActionIT
  2> REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestChangePolicyActionIT.test change policy API should only apply to indices in the state filter" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=sr-Latn-BA -Dtests.timezone=Etc/GMT+12 -Druntime.java=14
  2> org.opensearch.client.ResponseException: method [POST], host [https://localhost:9200], URI [.opendistro-ism-config/_search], status line [HTTP/1.1 404 Not Found]
    {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [.opendistro-ism-config]","index":".opendistro-ism-config","resource.id":".opendistro-ism-config","resource.type":"index_or_alias","index_uuid":"_na_"}],"type":"index_not_found_exception","reason":"no such index [.opendistro-ism-config]","index":".opendistro-ism-config","resource.id":".opendistro-ism-config","resource.type":"index_or_alias","index_uuid":"_na_"},"status":404}
        at __randomizedtesting.SeedInfo.seed([FC0793C0602FDFDE:F623A689A9CB1EE2]:0)
        at org.opensearch.client.RestClient.convertResponse(RestClient.java:344)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:314)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:289)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest(TestHelpers.kt:106)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest$default(TestHelpers.kt:95)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase.getManagedIndexConfig(IndexStateManagementRestTestCase.kt:317)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase$getExistingManagedIndexConfig$1.invoke(IndexStateManagementRestTestCase.kt:376)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase$getExistingManagedIndexConfig$1.invoke(IndexStateManagementRestTestCase.kt:94)
        at org.opensearch.indexmanagement.TestHelpersKt.waitFor(TestHelpers.kt:140)
        at org.opensearch.indexmanagement.TestHelpersKt.waitFor$default(TestHelpers.kt:134)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase.getExistingManagedIndexConfig(IndexStateManagementRestTestCase.kt:375)
        at org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestChangePolicyActionIT.test change policy API should only apply to indices in the state filter(RestChangePolicyActionIT.kt:455)
  2> REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestChangePolicyActionIT.test starting from a specific state using change policy API" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=sr-Latn-BA -Dtests.timezone=Etc/GMT+12 -Druntime.java=14
  2> org.opensearch.client.ResponseException: method [POST], host [https://localhost:9200], URI [.opendistro-ism-config/_search], status line [HTTP/1.1 404 Not Found]
    {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [.opendistro-ism-config]","index":".opendistro-ism-config","resource.id":".opendistro-ism-config","resource.type":"index_or_alias","index_uuid":"_na_"}],"type":"index_not_found_exception","reason":"no such index [.opendistro-ism-config]","index":".opendistro-ism-config","resource.id":".opendistro-ism-config","resource.type":"index_or_alias","index_uuid":"_na_"},"status":404}
        at __randomizedtesting.SeedInfo.seed([FC0793C0602FDFDE:B93EDED510B9F73F]:0)
        at org.opensearch.client.RestClient.convertResponse(RestClient.java:344)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:314)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:289)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest(TestHelpers.kt:106)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest$default(TestHelpers.kt:95)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase.getManagedIndexConfig(IndexStateManagementRestTestCase.kt:317)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase$getExistingManagedIndexConfig$1.invoke(IndexStateManagementRestTestCase.kt:376)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase$getExistingManagedIndexConfig$1.invoke(IndexStateManagementRestTestCase.kt:94)
        at org.opensearch.indexmanagement.TestHelpersKt.waitFor(TestHelpers.kt:140)
        at org.opensearch.indexmanagement.TestHelpersKt.waitFor$default(TestHelpers.kt:134)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase.getExistingManagedIndexConfig(IndexStateManagementRestTestCase.kt:375)
        at org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestChangePolicyActionIT.access$getExistingManagedIndexConfig(RestChangePolicyActionIT.kt:61)
        at org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestChangePolicyActionIT$test starting from a specific state using change policy API$1.invoke(RestChangePolicyActionIT.kt:539)
        at org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestChangePolicyActionIT$test starting from a specific state using change policy API$1.invoke(RestChangePolicyActionIT.kt:61)
        at org.opensearch.indexmanagement.TestHelpersKt.waitFor(TestHelpers.kt:140)
        at org.opensearch.indexmanagement.TestHelpersKt.waitFor$default(TestHelpers.kt:134)
        at org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestChangePolicyActionIT.test starting from a specific state using change policy API(RestChangePolicyActionIT.kt:539)
  2> REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestChangePolicyActionIT.test allowing change policy to happen in middle of state if same state structure" -Dtests.seed=FC0793C0602FDFDE -Dtests.security.manager=false -Dtests.locale=sr-Latn-BA -Dtests.timezone=Etc/GMT+12 -Druntime.java=14
  2> org.opensearch.client.ResponseException: method [GET], host [https://localhost:9200], URI [/_plugins/_ism/explain/restchangepolicyactionit_safe-000001], status line [HTTP/1.1 404 Not Found]
    {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [restchangepolicyactionit_safe-000001]","index":"restchangepolicyactionit_safe-000001","resource.id":"restchangepolicyactionit_safe-000001","resource.type":"index_or_alias","index_uuid":"_na_"}],"type":"index_not_found_exception","reason":"no such index [restchangepolicyactionit_safe-000001]","index":"restchangepolicyactionit_safe-000001","resource.id":"restchangepolicyactionit_safe-000001","resource.type":"index_or_alias","index_uuid":"_na_"},"status":404}
        at __randomizedtesting.SeedInfo.seed([FC0793C0602FDFDE:203FEA3AF6611A15]:0)
        at org.opensearch.client.RestClient.convertResponse(RestClient.java:344)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:314)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:289)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest(TestHelpers.kt:130)
        at org.opensearch.indexmanagement.TestHelpersKt.makeRequest$default(TestHelpers.kt:120)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase.getExplainManagedIndexMetaData(IndexStateManagementRestTestCase.kt:595)
        at org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestChangePolicyActionIT.access$getExplainManagedIndexMetaData(RestChangePolicyActionIT.kt:61)
        at org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestChangePolicyActionIT$test allowing change policy to happen in middle of state if same state structure$1.invoke(RestChangePolicyActionIT.kt:586)
        at org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestChangePolicyActionIT$test allowing change policy to happen in middle of state if same state structure$1.invoke(RestChangePolicyActionIT.kt:61)
        at org.opensearch.indexmanagement.TestHelpersKt.waitFor(TestHelpers.kt:140)
        at org.opensearch.indexmanagement.TestHelpersKt.waitFor$default(TestHelpers.kt:134)
        at org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestChangePolicyActionIT.test allowing change policy to happen in middle of state if same state structure(RestChangePolicyActionIT.kt:586)
  2> NOTE: leaving temporary files on disk at: /tmp/tmp0q50wr1x/index-management/build/testrun/integTest/temp/org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestChangePolicyActionIT_FC0793C0602FDFDE-001
  2> NOTE: test params are: codec=Asserting(Lucene87): {}, docValues:{}, maxPointsInLeafNode=213, maxMBSortInHeap=6.196754616671964, sim=Asserting(RandomSimilarity(queryNorm=true): {}), locale=sr-Latn-BA, timezone=Etc/GMT+12
  2> NOTE: Linux 4.14.243-185.433.amzn2.x86_64 amd64/AdoptOpenJDK 14.0.2 (64-bit)/cpus=16,threads=1,free=263944152,total=536870912
  2> NOTE: All tests run in this JVM: [ManagedIndexRunnerIT, ActionRetryIT, ActionTimeoutIT, AllocationActionIT, CloseActionIT, ForceMergeActionIT, IndexPriorityActionIT, IndexStateManagementHistoryIT, OpenActionIT, ReadOnlyActionIT, ReadWriteActionIT, ReplicaCountActionIT, RolloverActionIT, RollupActionIT, SnapshotActionIT, TransitionActionIT, ISMTemplateRestAPIIT, IndexStateManagementRestApiIT, RestAddPolicyActionIT, RestChangePolicyActionIT]
 
229 tests completed, 12 failed, 6 skipped
```

**Comment by ohltyler** [comment]
From @downsrob , there has been a lot of efforts in 1.3.0.0 to address flaky integ tests, including #211, #212, #247. Closing for now, and will open a new issue if new failed tests are seen in 1.3+.

**Comment by ohltyler** [comment]
One failure seen in 1.3.0:
```
REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestAddPolicyActionIT.test index pattern matching blocked indices" -Dtests.seed=AE1592C20572C6EE -Dtests.security.manager=false -Dtests.locale=ar-DZ -Dtests.timezone=Turkey -Druntime.java=14
  2> org.opensearch.client.WarningFailureException: method [GET], host [http://localhost:9200], URI [/.opendistro_security/_settings?flat_settings=true], status line [HTTP/1.1 200 OK]
    Warnings: [this request accesses system indices: [.opendistro_security], but in a future major version, direct access to system indices will be prevented by default]
    {".opendistro_security":{"settings":{"index.creation_date":"1643760352714","index.hidden":"true","index.number_of_replicas":"1","index.number_of_shards":"1","index.plugins.index_state_management.rollover_alias":null,"index.provided_name":".opendistro_security","index.uuid":"4on9uccLQ7uZ4P_CRGolUA","index.version.created":"135247827","index.write.wait_for_active_shards":"1"}}}
        at __randomizedtesting.SeedInfo.seed([AE1592C20572C6EE:7320E879F4F97B91]:0)
        at org.opensearch.client.RestClient.convertResponse(RestClient.java:340)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:314)
        at org.opensearch.client.RestClient.performRequest(RestClient.java:289)
        at org.opensearch.test.rest.OpenSearchRestTestCase.getIndexSettings(OpenSearchRestTestCase.java:1028)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase.access$getIndexSettings$s-249293656(IndexStateManagementRestTestCase.kt:73)
        at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase.getUuid(IndexStateManagementRestTestCase.kt:527)
        at org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestAddPolicyActionIT.test index pattern matching blocked indices(RestAddPolicyActionIT.kt:210)
  2> NOTE: leaving temporary files on disk at: /tmp/tmp6oqw5gcr/index-management/build/testrun/integTest/temp/org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestAddPolicyActionIT_AE1592C20572C6EE-001
  2> NOTE: test params are: codec=Lucene87, sim=Asserting(RandomSimilarity(queryNorm=false): {}), locale=ar-DZ, timezone=Turkey
  2> NOTE: Linux 4.14.243-185.433.amzn2.x86_64 amd64/AdoptOpenJDK 14.0.2 (64-bit)/cpus=16,threads=1,free=503522336,total=536870912
  2> NOTE: All tests run in this JVM: [ManagedIndexRunnerIT, MigrationServicesIT, ActionRetryIT, ActionTimeoutIT, AllocationActionIT, CloseActionIT, DeleteActionIT, ForceMergeActionIT, IndexPriorityActionIT, IndexStateManagementHistoryIT, OpenActionIT, ReadOnlyActionIT, ReadWriteActionIT, ReplicaCountActionIT, RolloverActionIT, RollupActionIT, SnapshotActionIT, TransitionActionIT, ISMTemplateRestAPIIT, IndexStateManagementRestApiIT, RestAddPolicyActionIT]

239 tests completed, 1 failed, 6 skipped
```

**Comment by thalurur** [comment]
@ohltyler is this seen consistently when running the test suite? 

> One failure seen in 1.3.0:
> 
> ```
> REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestAddPolicyActionIT.test index pattern matching blocked indices" -Dtests.seed=AE1592C20572C6EE -Dtests.security.manager=false -Dtests.locale=ar-DZ -Dtests.timezone=Turkey -Druntime.java=14
>   2> org.opensearch.client.WarningFailureException: method [GET], host [http://localhost:9200], URI [/.opendistro_security/_settings?flat_settings=true], status line [HTTP/1.1 200 OK]
>     Warnings: [this request accesses system indices: [.opendistro_security], but in a future major version, direct access to system indices will be prevented by default]
>     {".opendistro_security":{"settings":{"index.creation_date":"1643760352714","index.hidden":"true","index.number_of_replicas":"1","index.number_of_shards":"1","index.plugins.index_state_management.rollover_alias":null,"index.provided_name":".opendistro_security","index.uuid":"4on9uccLQ7uZ4P_CRGolUA","index.version.created":"135247827","index.write.wait_for_active_shards":"1"}}}
>         at __randomizedtesting.SeedInfo.seed([AE1592C20572C6EE:7320E879F4F97B91]:0)
>         at org.opensearch.client.RestClient.convertResponse(RestClient.java:340)
>         at org.opensearch.client.RestClient.performRequest(RestClient.java:314)
>         at org.opensearch.client.RestClient.performRequest(RestClient.java:289)
>         at org.opensearch.test.rest.OpenSearchRestTestCase.getIndexSettings(OpenSearchRestTestCase.java:1028)
>         at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase.access$getIndexSettings$s-249293656(IndexStateManagementRestTestCase.kt:73)
>         at org.opensearch.indexmanagement.indexstatemanagement.IndexStateManagementRestTestCase.getUuid(IndexStateManagementRestTestCase.kt:527)
>         at org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestAddPolicyActionIT.test index pattern matching blocked indices(RestAddPolicyActionIT.kt:210)
>   2> NOTE: leaving temporary files on disk at: /tmp/tmp6oqw5gcr/index-management/build/testrun/integTest/temp/org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestAddPolicyActionIT_AE1592C20572C6EE-001
>   2> NOTE: test params are: codec=Lucene87, sim=Asserting(RandomSimilarity(queryNorm=false): {}), locale=ar-DZ, timezone=Turkey
>   2> NOTE: Linux 4.14.243-185.433.amzn2.x86_64 amd64/AdoptOpenJDK 14.0.2 (64-bit)/cpus=16,threads=1,free=503522336,total=536870912
>   2> NOTE: All tests run in this JVM: [ManagedIndexRunnerIT, MigrationServicesIT, ActionRetryIT, ActionTimeoutIT, AllocationActionIT, CloseActionIT, DeleteActionIT, ForceMergeActionIT, IndexPriorityActionIT, IndexStateManagementHistoryIT, OpenActionIT, ReadOnlyActionIT, ReadWriteActionIT, ReplicaCountActionIT, RolloverActionIT, RollupActionIT, SnapshotActionIT, TransitionActionIT, ISMTemplateRestAPIIT, IndexStateManagementRestApiIT, RestAddPolicyActionIT]
> 
> 239 tests completed, 1 failed, 6 skipped
> ```

**Comment by ohltyler** [comment]
I'm seeing similar `WarningFailureException`s in AD plugin - common theme is all of these are happening when calling [indexExists() utility fn](https://github.com/opensearch-project/OpenSearch/blob/main/test/framework/src/main/java/org/opensearch/test/rest/OpenSearchRestTestCase.java#L1057) provided by OpenSearch, when accessing system indices (prefixed with a `.`). When looking at the [exception](https://github.com/opensearch-project/OpenSearch/blob/6f893ed1cd772abbe054f419869e54a208c12034/client/rest/src/main/java/org/opensearch/client/WarningFailureException.java) itself, appears it is thrown when the client would return a warning that would be interpreted as a failure if strict deprecation mode was set to `true`. In the failure itself, it shows the warning pertaining to system index access: `this request accesses system indices`.

However, not seeing this when running any versions previous of 1.3.0. Root cause must be related to something in the core OpenSearch test structure that's either setting this deprecation mode to true by default, or enabling it some other way.

Related old commit (may not be useful): [link](https://github.com/opensearch-project/OpenSearch/commit/75abb5b8a6b7b3cb9e54cd40c4afaa635fda2e87)

**Comment by ohltyler** [comment]
@thalurur actually just posted my findings so far above ^. I can't seem to replicate on other versions, or locally. Still looking. Have you seen this error before?

**Comment by thalurur** [comment]
@ohltyler got it, yes I have seen this error in older versions of ES but not in OS versions when run locally. 
As you stated the OS test framework creates RestClient with warning check enabled by default, and if there are warnings thrown the test will be failed. We have a mechanism to override this in some of our tests by [overriding the request headers](https://github.com/opensearch-project/index-management/blob/main/src/test/kotlin/org/opensearch/indexmanagement/TestHelpers.kt#L78), but we are not using this to get the index settings in above test. We can refactor if this is an issue. But curious why this is not failing locally but on jenkins

**Comment by ohltyler** [comment]
Got it, I see. I'm looking into what's different in Jenkins - I'm running the same cmd on a local host to see if I get those same errors and will report back.

Can you point me to where the warning check is enabled by default? I noticed that in AD's case, we actually enable it ourselves (see [here](https://github.com/opensearch-project/anomaly-detection/blob/main/src/test/java/org/opensearch/ad/ODFERestTestCase.java#L100))

**Comment by ohltyler** [comment]
I'm able to replicate the same errors exactly (including the related AD errors) by checking out the opensearch-build repo and testing a built 1.3.0 distribution:
```
./test.sh integ-test manifests/1.3.0/opensearch-1.3.0-test.yml --paths opensearch=https://ci.opensearch.org/ci/dbc/Playground/ohltyler-distribution-build-opensearch/1.3.0/43/linux/x64
```

I'm going to test the individual plugin integ tests (e.g., `./gradlew integTest`) on 1.3.0 to see if I can reproduce.

**Comment by thalurur** [comment]
@ohltyler its in [OpenSearchRestTestCase](https://github.com/opensearch-project/OpenSearch/blob/1.x/test/framework/src/main/java/org/opensearch/test/rest/OpenSearchRestTestCase.java#L808) (initClient method is used to initialize the clients which calls buildClient)

> Got it, I see. I'm looking into what's different in Jenkins - I'm running the same cmd on a local host to see if I get those same errors and will report back.
> 
> Can you point me to where the warning check is enabled by default? I noticed that in AD's case, we actually enable it ourselves (see [here](https://github.com/opensearch-project/anomaly-detection/blob/main/src/test/java/org/opensearch/ad/ODFERestTestCase.java#L100))

**Comment by ohltyler** [comment]
Individual plugin tests working fine.

I'm able to reproduce the individual plugin failures when I fully replicate the infra build test workflow for the specific component. Note that for ISM, this failure is only happening on security-disabled scenario (as seen in Jenkins + when I was running locally):
1. checkout opensearch-build repo
2. checkout ISM plugin repo
3. download & extract dist build tarball (e.g. https://ci.opensearch.org/ci/dbc/Playground/ohltyler-distribution-build-opensearch/1.3.0/43/linux/x64/dist/opensearch/opensearch-1.3.0-linux-x64.tar.gz)
4. append `plugins.security.disabled: true` to `config/opensearch.yml`
5. run cluster via `./opensearch-tar-install.sh`
6. `cd` to ISM plugin, run tests by referencing default integ-test script: `../opensearch-build/scripts/default/integtest.sh -b localhost -p 9200 -s false -v 1.3.0`

So there seems to be some discrepancy in either the integ test script being used (see [here](https://github.com/opensearch-project/opensearch-build/blob/main/scripts/default/integtest.sh)) and/or the cluster configuration.

**Comment by ohltyler** [comment]
By running the [[raw cmd](https://github.com/opensearch-project/opensearch-build/blob/main/scripts/default/integtest.sh#L90)]() inside ISM plugin, can replicate the error too:
```
./gradlew integTest -Dopensearch.version=1.3.0 -Dbuild.snapshot=false -Dtests.rest.cluster="localhost:9200" -Dtests.cluster="localhost:9200" -Dtests.clustername="opensearch-integrationtest" -Dhttps=false -Duser=admin -Dpassword=admin --console=plain --tests "org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestAddPolicyActionIT.test index pattern matching blocked indices"
```

When running w/o the extra params, it passes:
```
./gradlew integTest --tests "org.opensearch.indexmanagement.indexstatemanagement.resthandler.RestAddPolicyActionIT.test index pattern matching blocked indices"
```

Note these aren't running against any remote cluster; the gradle task itself is provisioning a local cluster and running tests against it, so the cluster configuration is actually irrelevant here.

**Comment by ohltyler** [comment]
I'm confused because the tarball cluster is set to security-disabled, and when making the same `curl` cmd that the test is making, it returns an expected 404, since it is querying the security system index, which doesn't exist, since it's a security-disabled cluster:
```
curl localhost:9200/.opendistro_security/_settings?flat_settings=true
{"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [.opendistro_security]","index":".opendistro_security","resource.id":".opendistro_security","resource.type":"index_or_alias","index_uuid":"_na_"}],"type":"index_not_found_exception","reason":"no such index [.opendistro_security]","index":".opendistro_security","resource.id":".opendistro_security","resource.type":"index_or_alias","index_uuid":"_na_"},"status":404}%
```

But, in the failure trace during the integ test, it's returning details of some existing security index:
```
GET http://localhost:9200/.opendistro_security/_settings?flat_settings=true
[...]
{".opendistro_security":{"settings":{"index.creation_date":"1643914205423","index.hidden":"true","index.number_of_replicas":"1","index.number_of_shards":"1","index.plugins.index_state_management.rollover_alias":null,"index.provided_name":".opendistro_security","index.uuid":"YGieDOihR8CDr1pC46zowg","index.version.created":"135247827","index.write.wait_for_active_shards":"1"}}}
```

**Comment by ohltyler** [comment]
When checking the test client config, it looks like the strict deprecation mode is getting set to `true` and is the expectation when running the integ tests ([source](https://github.com/opensearch-project/index-management/blob/main/src/test/kotlin/org/opensearch/indexmanagement/ODFERestTestCase.kt#L167)). AD plugin has similar config setup ([source](https://github.com/opensearch-project/anomaly-detection/blob/main/src/test/java/org/opensearch/ad/ODFERestTestCase.java#L100)).

I'm still unable to track down the difference between a spun-up-on-the-fly local cluster vs. existing, running cluster that's causing this strict deprecation mode to get switched to `false` (and subsequently pass successfully) in the first scenario. But because the intentions are that this should be set to `true`, I think it should be best practice to write tests that don't trigger these deprecation warnings, where it can fail in certain scenarios.

I'm still going to look a bit more into why this is allowing to pass locally, but I think the test itself should be modified to not call a system index directly. I'll be making similar changes in AD.

cc @thalurur

**Comment by ohltyler** [comment]
@thalurur the closest I could find to a root cause is some issue with the client configs when interacting with a remote cluster. For AD, I found that the failing tests were all making requests to directly manipulate system indices using a default client with strict deprecation mode as `true`. I changed those to use an admin client which has strict deprecation mode set to `false`, as well as bypassing any permissions errors, if running against a security-enabled cluster. You can check out the [PR](https://github.com/opensearch-project/anomaly-detection/pull/372) if you're looking to do something similar here

**Comment by ohltyler** [comment]
@thalurur @downsrob lmk if I can help with any more info, but if we can get this taken care of soon that would be great, to help unblock 1.3.0 snapshot distribution build integ tests
